ReactiveCocoa/Swift/Atomic.swift

```diff
@@ -8,9 +8,56 @@
 
 import Foundation
 
+public protocol MutexType {
+	func lock()
+	func unlock()
+}
+
+public final class RecursiveLock: MutexType {
```

Comment:
I think we should, at least to keep code consistent. And doing that we can have a single type, instead this two, by making it configurable with a particular strategy, `Recursive` or `NonRecursive`, during initialization.